### PR TITLE
regression cleanup for branching core-v-verif

### PR DESCRIPTION
**Pull request by strichmo**
add release and dev versions of ci_check and rel_check
remove static compliance regression in favor of yaml-based compliance regression

Signed-off-by: Steve Richmond <Steve.Richmond@silabs.com>

### CODE REVIEW

**Comment by strichmo** [comment]
Please also note some of the name changes on Metrics.  
- Removed _regression suffix as it seems superfluous.
- The names should enable long-term tracking of a regression type (i.e. how has rel_check performed on master) over time
-

**Comment by strichmo** [comment]
Added commits to define YAML files for regressions on Github.

**Review by MikeOpenHWGroup — COMMENTED** [review]
Looks good to me @strichmo.  Before merging this in, I am going to ask @aimeepsutton to review as well.  (Mostly for her awareness.)

**Comment by MikeOpenHWGroup** [comment]
Hi @aimeepsutton, before I approve this merge, could you have a look at it?   Mostly I just want you to be aware of what is going on, but also as a double check that we are not heading off into the weeds.  After a discussion between me, @strichmo and @GTumbush, we settled on the following regressions:

- **cv32e40x_ci_check**: Run for PRs on dev branches (Must always pass). In general this yaml is minimal and not subject to large change during development.
- **cv32e40x_rel_check**: Initially created from "full" regression. Runs very short (or even single number) of expanded test set. Must always pass for release flows to/from master and rel branches (Therefore must not be effected by RTL instability.)
- **cv32e40x_full_covg**: The main full coverage regression. Run on schedules. Expected to have instability due to deeper RTL bugs. The regression we close coverage on.
- **cv32e40x_compliance**: Run on nightly schedule on dev branch. Currently not specifying for releases.
- **cv32e40x_benchmark**: Run on nightly schedule on dev branch. Stability can fluctuate.

File: .metrics.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
       }
     },
     {
-      "name":        "cv32e40p_ci_check_regression",
+      "name":        "cv32e40p_ci_check_dev",
```
Comment (MikeOpenHWGroup) [inline .metrics.json:35]: This name should be the default `name_of_ci_check_regression` used by `ci_check`.

**Review by MikeOpenHWGroup — CHANGES_REQUESTED** [review]
This all looks really good, but the default regression run by `ci_check` is no longer named in `.metrics.json`.  This make the testcase list generated null and only the compile step is executed:
```
$./ci_check --core cv32e40p -s xrun -p 
ci_check: topdir  : /home/mike/GitHubRepos/openhwgroup/core-v-verif/test_521
ci_check: core : cv32e40p
ci_check: name_of_ci_check_regression : cv32e40p_ci_check_regression
cd cv32e40p/sim/uvmt; make corev-dv CV_CORE=cv32e40p SIMULATOR=xrun; make comp CV_CORE=cv32e40p SIMULATOR=xrun USE_ISS=YES
$ 
```
Other than that, it all looks fine, and a local hacked copy to resolve the above runs clear for me.

**Comment by strichmo** [comment]
Thanks @MikeOpenHWGroup for the catch!   I fixed by changing the name to "dev" since each core should have a dev branch ci_check regression (i.e. the name should be <core>_ci_check_dev)

**Review by MikeOpenHWGroup — APPROVED** [review]

**Pull request merged**